### PR TITLE
Add course continue button, course progress and pricing table blocks

### PR DESCRIPTION
**Pull request by seothemes**
## Description

Adds:

- Course Continue Button block
- Course Progress block
- Pricing Table block

with Course ID settings allowing blocks to be used on other post types.

## How has this been tested?

Manually

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] This PR requires and contains at least one changelog file.
- [ ] My code has been tested.
- [ ] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/llms.template.functions.php [hunk 1]
```diff
@@ -649,6 +649,14 @@ function lifterlms_course_progress_bar( $progress, $link = false, $button = true
 	}
 }
 
+/**
+ * Outputs the html for a progress bar.
+ *
+ * @since Unknown
+ *
+ * @param  float  $percentage Percent completion.
+ * @return string
```
Comment (eri-trabiccolo) [inline includes/llms.template.functions.php]:
```suggestion
 * @param float $percentage Percent completion.
 * @return string
```

File: includes/shortcodes/class.llms.shortcodes.php [hunk 2]
```diff
@@ -471,28 +471,32 @@ public static function course_info( $atts ) {
...
 	 * @since 3.38.0 Added logic to display the bar only to enrolled user.
+	 * @since [version] Add support for `course_id` attribute and render preview in block editor.
 	 *
 	 * @param array $atts Associative array of shortcode attributes.
 	 * @return string
 	 */
 	public static function course_progress( $atts ) {
 
-		$course_id = self::get_course_id();
+		$course_id = $atts['course_id'] ?? self::get_course_id() ?? null;
```
Comment (eri-trabiccolo) [inline includes/shortcodes/class.llms.shortcodes.php:485]: This is good, but I'd move the block editor related stuff into the class.llms.shortcodes.blocks.php what do you think?
I'd use existing filters or add new ones if needed, but I wouldn't do the checks here.
Same for the pricing table and the course continue button.

File: includes/llms.template.functions.php [hunk 3]
```diff
@@ -663,81 +671,122 @@ function llms_get_progress_bar_html( $percentage ) {
...
 
-				</a>
+			/**
+			 * Filter the next lesson to link to when outputting the continue button.
+			 *
+			 * @since Unknown
+			 *
+			 * @param int             $lesson  Next lesson to link to.
+			 * @param LLMS_Post_Model $course  Course being displayed.
+			 * @param LLMS_Student    $student Student viewing the course.
```
Comment (eri-trabiccolo) [inline includes/llms.template.functions.php]:
```suggestion
			 * @param int          $lesson  Next lesson to link to.
			 * @param LLMS_Course  $course  Course being displayed.
			 * @param LLMS_Student $student Student viewing the course.
```

File: src/blocks/course-continue-button/index.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+// WordPress dependencies.
```
Comment (eri-trabiccolo) [inline src/blocks/course-continue-button/index.jsx:1]: This needs to be backward compatible. See what happens when editing a course created with a previous version:
screenshot-20230807_103907_465411364

Se what we already did with the course progress back in the past:
https://github.com/gocodebox/lifterlms-blocks/blob/trunk/src/js/blocks/course-progress/index.js#L85

File: src/blocks/pricing-table/index.jsx [hunk 5]
```diff
@@ -1,16 +1,13 @@
 // WordPress dependencies.
```
Comment (eri-trabiccolo) [inline src/blocks/pricing-table/index.jsx:1]: This doesn't seem to work correctly with memberships.
Editing a membership the correct pricing table is displayed, although in the Inspector I still see Course/Select course. While I should see Membership/Inherit from current. Right?
screenshot-20230807_104825_733504507

Also it should say "Select Membership" not "Select Course" when a membership is selected.

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]

File: includes/shortcodes/class.llms.shortcodes.php (hunk 2 above)
Comment (eri-trabiccolo) [reply includes/shortcodes/class.llms.shortcodes.php:485]: seems you missed this. There's a similar comment on the other PR.

File: includes/shortcodes/class.llms.shortcodes.php [hunk 6]
```diff
@@ -601,12 +606,26 @@ public static function pricing_table( $atts ) {
 
 		if ( ! empty( $atts['product'] ) && is_numeric( $atts['product'] ) ) {
 
-			// enqueue match height for height alignments.
+			$block_editor = llms_is_editor_block_rendering();
```
Comment (eri-trabiccolo) [inline includes/shortcodes/class.llms.shortcodes.php:609]: same as above.

File: src/blocks/pricing-table/index.jsx (hunk 5 above)
Comment (eri-trabiccolo) [reply src/blocks/pricing-table/index.jsx:1]: You seem to have missed this too

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]

File: src/blocks/pricing-table/index.jsx (hunk 5 above)
Comment (seothemes) [reply src/blocks/pricing-table/index.jsx:1]: @eri-trabiccolo this should be working correctly now

**Comment by brianhogg** [comment]
Would review and re-do if we want to add additional blocks.

**Pull request closed without merging**